### PR TITLE
feat: add ar_tag_based_localizer evaluation

### PR DESCRIPTION
**Pull request by SakodaShintaro**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description

Add `ar_tag_based_localizer` evaluator scenario.

## How to review this PR

Please follow the documentation for `ar_tag_based_localizer` scenario and confirm that it actually runs with a sample rosbag

### CODE REVIEW

File: driving_log_replayer/driving_log_replayer/ar_tag_based_localizer.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    def set_frame(self, msg: DiagnosticArray) -> dict:
+        include_target_status = False
+        for diag_status in msg.status:
+            if diag_status.name != Availability.TARGET_DIAG_NAME:
+                continue
+            include_target_status = True
+            values = {value.key: value.value for value in diag_status.values}
+            detection_count = int(values.get("Number of Detected AR Tags", "-1"))
+            print(f"Number of Detected AR Tags: {detection_count}")
```
Comment (hayato-m126) [inline driving_log_replayer/driving_log_replayer/ar_tag_based_localizer.py:37]: This print statement is unnecessary.

https://docs.astral.sh/ruff/rules/print/

Comment (SakodaShintaro) [reply driving_log_replayer/driving_log_replayer/ar_tag_based_localizer.py]: Sorry, the code for debugging was left.

Fixed in https://github.com/tier4/driving_log_replayer/pull/258/commits/860496bc274b95478286a1e2b25f1d66a9c649c3

File: docs/use_case/ar_tag_based_localizer.ja.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+### 入力 rosbag に含まれるべき topic
+
+| topic 名                                           | データ型                                      |
+| -------------------------------------------------- | --------------------------------------------- |
+| /sensing/camera/traffic_light/camera_info          | sensor_msgs/msg/CameraInfo                    |
+| /sensing/camera/traffic_light/image_raw/compressed | sensor_msgs/msg/CompressedImage               |
+| /sensing/imu/tamagawa/imu_raw                      | sensor_msgs/msg/Imu                           |
+| /vehicle/status/velocity_status                    | autoware_auto_vehicle_msgs/msg/VelocityReport |
+| /initialpose                                       | geometry_msgs/msg/PoseWithCovarianceStamped   |
```
Comment (hayato-m126) [inline docs/use_case/ar_tag_based_localizer.ja.md:76]: I don't think /initialpose is necessary.

If you describe `InitialPose` in your scenario, the following part calls the service of /initialpose every second when clock is greater than 0.
https://github.com/tier4/driving_log_replayer/blob/develop/driving_log_replayer/driving_log_replayer/evaluator.py#L135-L136

**Review by hayato-m126 — CHANGES_REQUESTED** [review]
I don't think /initialpose is needed for input_bag, please check.

File: docs/use_case/ar_tag_based_localizer.ja.md (hunk 2 above)
Comment (SakodaShintaro) [reply docs/use_case/ar_tag_based_localizer.ja.md]: It was my misunderstanding.
The `/initialpose` topic is not needed and DrivingLogReplayer works the same whether it is there or not.
So I removed it from the document.

**Review by hayato-m126 — APPROVED** [review]
LGTM
This PR works fine with the sample data.

```shell
test case 1 / 1 : use case: sample
--------------------------------------------------
TestResult: Passed
Passed: ArTagBasedLocalizer Availability (Success): Detected 1 AR tags
```

**Pull request merged**